avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerDe.java

```diff
@@ -66,14 +67,26 @@ protected void configureClientProperties(AbstractKafkaAvroSerDeConfig config) {
 
   private static String validateAndMaybeGetMockScope(List<String> urls) {
     String mockScope = null;
+    boolean sawANonMockUrl = false;
     for (String url : urls) {
-      final boolean isMock = url.startsWith("mock://");
+      final boolean isMock = url.startsWith(MOCK_URL_PREFIX);
+
```

Comment:
Maybe simplify the block below to

```
if (isMock) {
  if (mockScope != null) {
     ...
  }
  if (sawANonMockUrl) { 
    ...
  }
} else {
  sawANonMockUrl = true;
}

```

Reply:
I had something like that initially, but you still want to check that `mockScope != null` in the "not a mock" `else` block, so you wind up with a duplicated check that way.

Reply:
Specifically, you get:
```java
      if (isMock) {
        if (sawANonMockUrl) {
          throw new ConfigException(
                  "Cannot mix mock and real urls for 'schema.registry.url'. Got: " + urls
          );
        }
        mockScope = url.substring(MOCK_URL_PREFIX.length());
      } else {
        if (mockScope != null) {
          throw new ConfigException(
                  "Cannot mix mock and real urls for 'schema.registry.url'. Got: " + urls
          );
        }
        sawANonMockUrl = true;
      }
```

Reply:
Or, to consolidate all the checks into one nested block structure:

```java
      if (isMock) {
        if (sawANonMockUrl) {
          throw new ConfigException(
                  "Cannot mix mock and real urls for 'schema.registry.url'. Got: " + urls
          );
        } else if (mockScope != null) {
          throw new ConfigException(
                  "Only one mock scope is permitted for 'schema.registry.url'. Got: " + urls
          );
        } else {
          mockScope = url.substring(MOCK_URL_PREFIX.length());
        }
      } else {
        if (mockScope != null) {
          throw new ConfigException(
                  "Cannot mix mock and real urls for 'schema.registry.url'. Got: " + urls
          );
        } else {
          sawANonMockUrl = true;
        }
      }
```

Reply:
Ok, how about transforming the `List<String>` into a `Map<String, String>` and then you can check the map as follows:
```
if (map.size() > 1) {
      throw new ConfigException(
          "Cannot mix mock and real urls for 'schema.registry.url'. Got: " + urls
      );
    }
    List<String> scopes = map.get(MOCK_URL_PREFIX);
    String scope;
    if (scopes != null) {
      if (scopes.size() > 1) {
        throw new ConfigException(
            "Only one mock scope is permitted for 'schema.registry.url'. Got: " + urls
        );
      } else {
        scope = scopes.get(0);
      }
    }
    return scope;
```

Reply:
Ah, thanks for the idea! Actually, I realized that we can do the same thing without a map, just first building a list of defined mock scopes. But it's essentially what you proposed.

Much cleaner!